geos/wkb.go

```diff
@@ -41,6 +48,9 @@ func (d *wkbDecoder) decode(wkb []byte) (*Geometry, error) {
 }
 
 func (d *wkbDecoder) decodeHex(wkbHex string) (*Geometry, error) {
+	d.m.Lock()
```

Comment:
`d.m.Lock()` is taking a lock here, and then this function is calling `d.decode()` method, which will again try to take a lock on the same mutex at Line 36. This will cause it to wait forever.

Reply:
Force-pushed a fix for this.